README.rst

```diff
@@ -0,0 +1,3824 @@
+from a client or from a member where the keys do not exist, you
+effectively perform two network hops for each update: the first to
+retrieve the data and the second to update the mutated value.
+
+If you are doing the process described above, you should consider using
+entry processors. An entry processor executes a read and updates upon
+the member where the data resides. This eliminates the costly network
+hops described above.
+
+   NOTE: Entry processor is meant to process a single entry per call.
+   Processing multiple entries and data structures in an entry processor
+   is not supported as it may result in deadlocks on the server side.
+
+Hazelcast sends the entry processor to each cluster member and these
+members apply it to the map entries. Therefore, if you add more members,
+your processing completes faster.
+
+Processing Entries
+^^^^^^^^^^^^^^^^^^
+
+The ``Map`` class provides the following methods for entry processing:
+
+-  ``execute_on_key`` processes an entry mapped by a key.
+-  ``execute_on_keys`` processes entries mapped by a list of keys.
+-  ``execute_on_entries`` can process all entries in a map with a
+   defined predicate. Predicate is optional.
+
+In the Python client, an ``EntryProcessor`` should be
+``IdentifiedDataSerializable`` or ``Portable`` because the server should
+be able to deserialize it to process.
+
+The following is an example for ``EntryProcessor`` which is an
+``IdentifiedDataSerializable``.
+
+.. code:: python
+
+   from hazelcast.serialization.api import IdentifiedDataSerializable
+
+   class IdentifiedEntryProcessor(IdentifiedDataSerializable):
+       def __init__(self, value=None):
+           self.value = value
+       
+       def read_data(self, object_data_input):
+           self.value = object_data_input.read_utf()
+       
+       def write_data(self, object_data_output):
+           object_data_output.write_utf(self.value)
+       
+       def get_factory_id(self):
+           return 5
+           
+       def get_class_id(self):
+           return 1
+
+Now, you need to make sure that the Hazelcast member recognizes the
+entry processor. For this, you need to implement the Java equivalent of
+your entry processor and its factory, and create your own compiled class
+or JAR files. For adding your own compiled class or JAR files to the
+server’s ``CLASSPATH``, see the `Adding User Library to CLASSPATH
+section <#1212-adding-user-library-to-classpath>`__.
+
+The following is the Java equivalent of the entry processor in Python
+client given above:
+
+.. code:: java
+
+   import com.hazelcast.map.AbstractEntryProcessor;
+   import com.hazelcast.nio.ObjectDataInput;
+   import com.hazelcast.nio.ObjectDataOutput;
+   import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+   import java.io.IOException;
+   import java.util.Map;
+
+   public class IdentifiedEntryProcessor extends AbstractEntryProcessor<String, String> implements IdentifiedDataSerializable {
```

Comment:
nit: the `extends AbstractEntryProcessor<String, String> implements IdentifiedDataSerializable {` could be moved to the next line to keep the line shorter (GH shows a horizontal scrollbar for the current version).

Reply:
Also updated the code sample, `AbstractEntryProcessor` does not exist anymore

Reply:
Good catch. I've created https://github.com/hazelcast/hazelcast-nodejs-client/issues/661 to do the same in Node.js client.